002-routing-params-views/src/app/app.route.js

```diff
@@ -3,19 +3,25 @@
 
   angular
     .module('angularApp')
-    .config(routerConfig);
+    .config(routerConfig)
+    .run(ruterValidations);
 
   /** @ngInject */
   function routerConfig($stateProvider, $urlRouterProvider) {
-    $stateProvider
-      .state('home', {
-        url: '/',
-        templateUrl: 'app/states/home/home.html',
-        controller: 'HomeController',
-        controllerAs: 'homeCtrl'
-      });
-
     $urlRouterProvider.otherwise('/');
   }
 
+  function ruterValidations ($rootScope, $log, $state) {
+    var changeState = $rootScope.$on('$stateChangeStart', function (event, toState) {
+      var requireLogin = toState.data.requiresLogin;
+      if (requireLogin && localStorage.currentUser === 'undefined') {
+        event.preventDefault();
+        $state.go('home');
+      } else if (!requireLogin && localStorage.currentUser !== 'undefined') {
+        event.preventDefault();
+        $state.go('main');
+      }
```

Comment:
this logic should be extracted to a factory because, although in this case is very simple it may get complicated. (this wasn't part of the exercise but have it in mind)